common-service-core/src/main/java/org/zowe/apiml/util/CorsUtils.java

```diff
@@ -21,21 +21,29 @@
 import java.util.Map;
 import java.util.function.BiConsumer;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
-@RequiredArgsConstructor
 public class CorsUtils {
-    private static final List<String> allowedCorsHttpMethods;
+    private final List<String> allowedCorsHttpMethods;
     private final boolean corsEnabled;
     private final List<String> allowedOrigins;
     private static final Pattern gatewayRoutesPattern = Pattern.compile("apiml\\.routes\\.[^.]*\\.gateway\\S*");
-
     private static final List<String> CORS_ENABLED_ENDPOINTS = Arrays.asList("/*/*/gateway/**", "/gateway/*/*/**", "/gateway/version");
 
-    static {
-        allowedCorsHttpMethods = Collections.unmodifiableList(Arrays.asList(
-            HttpMethod.GET.name(), HttpMethod.HEAD.name(), HttpMethod.POST.name(),
-            HttpMethod.DELETE.name(), HttpMethod.PUT.name(), HttpMethod.OPTIONS.name()
-        ));
+    public CorsUtils(boolean corsEnabled, String corsAllowedMethods, List<String> allowedOrigins) {
```

Comment:
allowed origins are always null in our code, do we need them in the constructor?